modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/exp/agg/Accumulators.java

```diff
@@ -1158,6 +1191,71 @@ private String extractSeparator(Row row) {
         }
     }
 
+    /** */
+    private static class ArrayAggregateAccumulator<Row> extends AggAccumulator<Row> {
+        /** */
+        public ArrayAggregateAccumulator(AggregateCall aggCall, RowHandler<Row> hnd) {
+            super(aggCall, hnd);
+        }
+
+        /** {@inheritDoc} */
+        @Override public Object end() {
+            List<Object> result = new ArrayList<>(size());
+
+            for (Row row: this)
+                result.add(get(0, row));
```

Comment:
Do we really need to copy array? Is it reused somehow after `end()`?

Reply:
Buffer contains Rows (list of arrays)